packages/ubuntu_wsl_setup/integration_test/ubuntu_wsl_setup_test.dart

```diff
@@ -0,0 +1,379 @@
+      expect(ini.get('network', 'generateresolvconf'),
+          equals(config.networkGenerateresolvconf.toString()));
+    });
+  }
+
+  Future<void> verifyAdvancedConfig(WSLConfigurationAdvanced config) async {
+    await verifyIniFile('wsl.conf', (ini) {
+      expect(ini.get('automount', 'enabled'),
+          equals(config.automountEnabled.toString()));
+      expect(ini.get('automount', 'mountfstab'),
+          equals(config.automountMountfstab.toString()));
+    });
+
+    await verifyIniFile('ubuntu-wsl.conf', (ini) {
+      expect(ini.get('Interop', 'advancedipdetection'),
+          equals(config.interopAdvancedipdetection.toString()));
+      expect(ini.get('Interop', 'audiointegration'),
+          equals(config.interopAudiointegration.toString()));
+      expect(ini.get('Interop', 'guiintegration'),
+          equals(config.interopGuiintegration.toString()));
+      expect(ini.get('Motd', 'wslnewsenabled'),
+          equals(config.motdWSLnewsenabled.toString()));
+    });
+  }
+
+  testWidgets('basic setup', (tester) async {
+    app.main(<String>[]);
+    await waitForSubiquityServer();
+    await tester.pumpAndSettle();
+
+    await testSelectYourLanguage(tester, language: 'Français');
+    await tester.pumpAndSettle();
+
+    const profile = IdentityData(
+      realname: 'WSL User',
+      username: 'wsl-user',
+    );
+
+    await testProfileSetup(
+      tester,
+      profile,
+      password: 'password123',
+      confirmedPassword: 'password123',
+      showAdvancedOptions: false,
+    );
+    await tester.pumpAndSettle();
+
+    await testSetupComplete(tester, username: 'wsl-user');
+    await tester.pumpAndSettle();
+
+    await verifyLocale('fr_FR.UTF-8');
+    await verifyProfile(profile);
+  });
+
+  testWidgets('advanced setup', (tester) async {
+    app.main(<String>['--initial-route', Routes.profileSetup]);
+    await waitForSubiquityServer();
+    await tester.pumpAndSettle();
+
+    const profile = IdentityData(
+      realname: 'WSL User',
+      username: 'wsl-user',
+    );
+
+    await testProfileSetup(
+      tester,
+      profile,
+      password: 'password123',
+      confirmedPassword: 'password123',
+      showAdvancedOptions: true,
+    );
+    await tester.pumpAndSettle();
+
+    // NOTE: opposites of the default values to force writing the config
+    const config = WSLConfigurationBase(
+      automountRoot: '/tst/wsl',
+      automountOptions: '--test-wsl',
+      networkGeneratehosts: false,
+      networkGenerateresolvconf: false,
+    );
+
+    await testAdvancedSetup(tester, config);
+    await tester.pumpAndSettle();
+
+    await testSetupComplete(tester, username: 'wsl-user');
+    await tester.pumpAndSettle();
+
+    await verifyBaseConfig(config);
+  });
+
+  testWidgets('reconfiguration', (tester) async {
+    app.main(<String>['--reconfigure']);
+    await waitForSubiquityServer();
+    await tester.pumpAndSettle();
+
+    await testAdvancedSetup(tester);
```

Comment:
Mind adding a comment that this is testing the first screen (WSLConfigurationBase)?